### DISCUSSION TITLE
Visual stimulus not displayed with short presentation times (50/100ms)

### DISCUSSION THREAD

**Discussion by nmrzq** [jspsych/jsPsych#1085] [Q&A] [answered]
Hello everyone,

I am running a scene categorization task in which an image is presented briefly after which participants have to answer several questions. I had no difficulties running the experiment with a fairly long presentation time (1000 ms). However, when I reduce the presentation time to 50 or 100 ms, I encounter 10-20% of trials in which no stimulus is visible at all. And it seems like the problem is becoming more prevalent the shorter the presentation time is (more trials without stimuli in the 50 ms compared to the 100 ms version). 

Note: I use manual preloading of 240 images (+32 practice images) which might create a performance bottleneck(?).

I am using Firefox 81.0, but colleagues reported the same problems with Safari and Chrome. I quickly tested this on Microsoft Edge, but there I had cut-off stimuli on some trials (only top 20-50% of stimuli presented). 

Any help is appreciated!

Thanks in advance,
Sandro

**Comment by becky-gilbert** [comment, marked as answer]
Hi @nmrzq! Thanks for sharing your live experiment and code. I ran your experiment and I think I see what you mean about the images not appearing. My first guess is that the problem is that the images aren't preloading successfully, so I opened the [network tab](https://developers.google.com/web/tools/chrome-devtools/network#open) in the browser's developer tools to see when the images are loaded. I did see lots of preloaded images that start with a number, e.g. "13_o1.jpg", but then when I started your task, I saw some images that start with a "p" that were not preloaded, e.g. "p11_s.jpg". Since those "p" images aren't preloaded, the browser is requesting them at the start of the trial, and they're not transferred back from the server in time to be shown before the trial ends. This may not be the only issue, but I wonder whether preloading those images fixes the problem?

**Reply by nmrzq** [reply]
Hi @becky-gilbert, thank you for your reply and taking the time to look at my experiment! You are right, I simply forgot to specify the stimuli for the practice trials ("px_y.jpg") in the preloading list (sometimes it's the little things...). Including them solved the problem. Thank you again for pointing that out and linking to the network tab, I didn't know of that feature!
Best,
Sandro
